cmp/filter/impl/Option.ts

```diff
@@ -70,7 +70,12 @@ export function fieldFilterOption({filter, fieldSpec, isExact = false}): FilterC
         displayValue = filter.op === '!=' ? 'not blank' : 'blank';
     } else {
         displayOp = filter.op;
-        fieldType = fieldType === 'tags' ? 'string' : fieldType;
+        fieldType =
+            fieldType === 'tags'
+                ? 'string'
+                : fieldSpec.isMismatchedDateFieldType
+                ? 'date'
+                : fieldType;
```

Comment:
If we use localDate as the fieldType, we end up passing a date to the localDate case in Field's parseFieldValue function and would then need to use LocalDate.from(val) instead of LocalDate.get(val).